### PR TITLE
Add functionality to Edit Tasks

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**Description of Changes**

- Added "Edit" button to each task in TaskList component
- Added editFormState state to manage which task is being edited.
- Implemented conditional inline editing form for tasks that are in "edit" mode
- Added updateTask function in MainContainer to handle task updates 

https://github.com/user-attachments/assets/908102df-2852-4255-89ba-6b6d348d728d

### CODE REVIEW

**Comment by ishmam-iqbal-sazim** [comment]
Could you add screen recordings to the PR description to reflect the changes made?

Also, there seems to be a merge commit in your commit history, not sure what happened there. Let's sync the next time you merge a branch and we can figure it out.

**Comment by yamin-faruque-sazim** [comment]
I think the merge commit might be due to the fact that, I mistakenly git pulled the latest changes of remote to my current working branch, resolved the conflicts and then pushed, instead of pulling it to main and then rebasing my current branch to main.

File: src/Pages/MainPage/Components/TaskList.tsx [hunk 1]
```diff
@@ -36,6 +70,25 @@ const TaskList: React.FC<TaskListProps> = ({
     toggleCompletedTask(id);
   };
 
+  const handleEditChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value } = e.target;
+    setEditFormState((prev) => ({ ...prev, [name]: value }));
+  };
```
Comment (ishmam-iqbal-sazim) [inline src/Pages/MainPage/Components/TaskList.tsx:78]: You can look into usage of form and validation libraries such as react hook form and zod.

Can create a ticket to hook that up later

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Comment by yamin-faruque-sazim** [comment]
Also added video recording of the feature

**Pull request merged**